kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/GCSToBQWriter.java

```diff
@@ -108,23 +109,28 @@ public void writeRows(List<RowToInsert> rows,
     }
 
     int retryCount = 0;
-    boolean exceptionsOccurred;
+    boolean success = false;
     do {
       if (retryCount > 0) {
         waitRandomTime();
       }
-      exceptionsOccurred = false;
-      // Perform GCS Upload and BQ Transfer
+      // Perform GCS Upload
       try {
         uploadRowsToGcs(rows, blobInfo);
-      } catch (ConnectException ce) {
-        exceptionsOccurred = true;
+        success = true;
+      } catch (StorageException se) {
         logger.warn("Exceptions occurred for table {}, attempting retry", tableId.getTable());
       }
       retryCount++;
-    } while (exceptionsOccurred && (retryCount < retries));
+    } while (!success && (retryCount < retries));
```

Comment:
Possible to move this to a normal while() rather than do/while as a best-practices thing?